### PR TITLE
nixos/prometheus/alertmanager: use DynamicUser instead of nobody

### PR DESCRIPTION
**Pull request by basvandijk**
###### Motivation for this change

See issue #55370.

@arianvp care to review?

The following which includes a test for `alertmanager` passes but I'll ask ofborg to confirm:

`nix-build nixos/release.nix -A tests.prometheus.x86_64-linux`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by basvandijk** [comment]
@GrahamcOfBorg test prometheus

File: nixos/modules/services/monitoring/prometheus/alertmanager.nix [hunk 1]
```diff
@@ -152,10 +136,8 @@ in {
         wantedBy = [ "multi-user.target" ];
         after    = [ "network.target" ];
         serviceConfig = {
-          User = cfg.user;
-          Group = cfg.group;
           Restart  = "always";
-          PrivateTmp = true;
+          DynamicUser = true;
```
Comment (arianvp) [inline nixos/modules/services/monitoring/prometheus/alertmanager.nix]: We should be able to keep PrivateTmp here I think

**Review by arianvp — CHANGES_REQUESTED** [review]
We should add an `mkRemovedOptionModule`  entry to `nixos/modules/rename.nix` so people get warned if they used the `user` and `group` options.

Apart from that and my inline comments, lgtm

**Comment by arianvp** [comment]
Or we could just keep the `user` and `group` options. Though I am not sure if that adds anything in the case of `DynamicUser`. Pinging @flokli for opinion

File: nixos/modules/services/monitoring/prometheus/alertmanager.nix (hunk 1 above)
Comment (basvandijk) [reply nixos/modules/services/monitoring/prometheus/alertmanager.nix]: We can remove it because `DynamicUser` implies `PrivateTmp`.

**Comment by basvandijk** [comment]
> Or we could just keep the user and group options. 

IIUC the `User` and `Group` systemd options become necessary when using `DynamicUser`.

**Comment by basvandijk** [comment]
> We should add an `mkRemovedOptionModule` entry to `nixos/modules/rename.nix`

Thanks, will do that.

**Review by arianvp — APPROVED** [review]

**Comment by arianvp** [comment]
* Could you add a comment in the removeOptions thing that tells why they were removed? 
* Could you add an entry to the change log?
Apart from that lgtm

**Comment by basvandijk** [comment]
Done. Thank you for the review. Will merge after checks are green.

**Pull request merged**